python/amici/setup.template.py

```diff
@@ -25,22 +32,22 @@ def getAmiciLibs():
             #'klu', 'colamd', 'btf', 'amd', 'suitesparseconfig'
             ]
 
-# Find HDF5
-import pkgconfig
-h5pkgcfg = pkgconfig.parse("hdf5")
-
 cxx_flags = ['-std=c++11']
-#linker_flags = ['${BLAS_LIBRARIES}']
 linker_flags = []
-if 'ENABLE_GCOV_COVERAGE' in os.environ and os.environ['ENABLE_GCOV_COVERAGE'] == 'TRUE':
-    cxx_flags.extend(['-g', '-O0',  '--coverage'])
-    linker_flags.append('--coverage')
 
-libraries = [*getAmiciLibs(),
-             'cblas',# TODO generic BLAS
-             'hdf5_hl_cpp', 'hdf5_hl', 'hdf5_cpp', 'hdf5']
-sources = ['swig/TPL_MODELNAME.i', *getModelSources()]
+addCoverageFlagsIfRequired(cxx_flags, linker_flags)
+addDebugFlagsIfRequired(cxx_flags, linker_flags)
 
+h5pkgcfg = getHdf5Config()
+
+blaspkgcfg = getBlasConfig()
+linker_flags.extend(blaspkgcfg.get('extra_link_args', []))
+
+libraries = [*getAmiciLibs(), *blaspkgcfg['libraries']]
+if hdf5_enabled:
+    libraries.extend(['hdf5_hl_cpp', 'hdf5_hl', 'hdf5_cpp', 'hdf5'])
+
+sources = ['swig/TPL_MODELNAME.i', *getModelSources()]
```

Comment:
honestly can't tell if all flags and options are set correctly, but looks reasonable